mmv1/third_party/terraform/utils/privateca_utils.go

```diff
@@ -13,6 +14,234 @@ import (
 // deleted on the API-side. This flattener creates default objects for sub-objects that match this pattern
 // to fix perma-diffs on default-only objects. For this to work all objects that are flattened from nil to
 // their default object *MUST* be set in the user's config, so they are all marked as Required in the schema.
+//
+// This file also contains shared expanders between the above resources. The expanders are required in order
+// to handle the optional primitive field in CaOptions. By adding a virtual field, the expander can distinguish
+// between an unset primitive field and a set primitive field with a default value.
+
+// Expander utilities
+
+func expandPrivatecaCertificateConfigX509ConfigCaOptions(v interface{}, d TerraformResourceData, config *Config) (interface{}, error) {
+	// Virtual fields to distinguish between unset booleans and booleans set with a default value.
+	// Unset is_ca or unset max_issuer_path_length either allow any values for these fields when
+	// used in an issuance policy, or allow the API to use default values when used in a
+	// certificate config. A default value of is_ca=false means that issued certificates cannot
+	// be CA certificates. A default value of max_issuer_path_length=0 means that the CA cannot
+	// issue CA certificates.
+
+	// includeIsCa can be nil for Certificate Authority resources
+	includeIsCa := d.Get("include_is_ca")
+	includePathLength := d.Get("include_max_path_length")
+
+	if v == nil {
+		if includeIsCa != nil && includeIsCa.(bool) {
```

Comment:
Is it ever possible for `includeIsCa` to be nil? Because it's a boolean it should default to false and never be nil